frontend/packages/process-editor/src/hooks/useBpmnEditor.test.tsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import { renderHook, waitFor } from '@testing-library/react';
+import { useBpmnEditor } from './useBpmnEditor';
+import { BpmnContextProvider, useBpmnContext } from '../contexts/BpmnContext';
+import { BpmnApiContextProvider } from '../contexts/BpmnApiContext';
+import { useBpmnModeler } from './useBpmnModeler';
+
+const bpmnDetailsMock = {
+  id: 'testId',
+  type: 'bpmn:Task',
+};
+
+class BpmnModelerMockImpl {
+  public readonly _currentEventName: string;
+  private readonly eventBus: any;
+
+  constructor(currentEventName: string) {
+    this._currentEventName = currentEventName;
+    this.eventBus = {
+      _currentEventName: this._currentEventName,
+      on: this.on,
+    };
+  }
+  on(eventName: string, listener: (event: any) => void) {
+    if (eventName === this._currentEventName) {
+      listener({ element: 'someElement' });
+    }
+  }
+  get(elementName: string) {
+    if (elementName === 'eventBus') {
+      return this.eventBus;
+    }
+  }
+}
+
+jest.mock('../utils/hookUtils', () => ({
+  getBpmnEditorDetailsFromBusinessObject: jest.fn().mockReturnValue({
+    id: 'testId',
+    type: 'bpmn:Task',
+  }),
+}));
+
+jest.mock('../contexts/BpmnConfigPanelContext', () => ({
+  useBpmnConfigPanelFormContext: jest.fn(() => ({
+    metaDataFormRef: { current: null },
+    resetForm: jest.fn(),
+  })),
+}));
+
+jest.mock('../contexts/BpmnContext', () => ({
+  ...jest.requireActual('../contexts/BpmnContext'),
+  useBpmnContext: jest.fn(() => ({
+    getUpdatedXml: jest.fn(),
+    modelerRef: { current: null },
+    setBpmnDetails: jest.fn(),
+  })),
+}));
+
+const setBpmnDetailsMock = jest.fn();
+const overrideUseBpmnContext = () => {
+  (useBpmnContext as jest.Mock).mockReturnValue({
+    getUpdatedXml: jest.fn(),
+    modelerRef: { current: null },
+    setBpmnDetails: setBpmnDetailsMock,
+  });
+};
+
+jest.mock('./useBpmnModeler', () => ({
+  useBpmnModeler: jest.fn().mockReturnValue({
+    getModeler: jest.fn(),
+  }),
+}));
+
+const overrideUseBpmnModeler = (currentEventName: string) => {
+  (useBpmnModeler as jest.Mock).mockReturnValue({
+    getModeler: () => new BpmnModelerMockImpl(currentEventName),
+  });
+};
+
+describe('useBpmnEditor', () => {
+  afterEach(jest.clearAllMocks);
+  it('should call saveBpmn when "commandStack.changed" event is triggered on modelerInstance', async () => {
+    const saveBpmnMock = jest.fn();
```

Comment:
Some of these functions are used in many of the tests in this file, could we move them outside the test and re-use them in all tests? 😄